dev-utils/karma.js

```diff
@@ -127,6 +127,12 @@ function prepareConfig(defaultConfig) {
       ') Elastic Stack ' +
       process.env.STACK_VERSION
 
+    defaultConfig.plugins.push('karma-junit-reporter')
```

Comment:
sorry my bad didnt review it properly. we have to add the check if coverage is enabled before doing this. Otherwise reports will be created for normal tests as well. 

```js
if (defaultConfig.coverage) {
 // code goes here
}
```

Reply:
Uhm, I'm a bit confused now. It seems there were above 1000 tests and since I rebased with the master branch the number of reported tests have dramatically reduced to less than 100.

- Previously - see [here](https://apm-ci.elastic.co/job/apm-agent-rum/job/apm-agent-rum-mbp/job/PR-323/2/testReport/)
- Now - see [here](https://apm-ci.elastic.co/job/apm-agent-rum/job/apm-agent-rum-mbp/job/PR-323/5/testReport/)

is that the expected behavior?

Reply:
@vigneshshanmugam 
>> Otherwise reports will be created for normal tests as well.
Not sure what you meant, the suggested approach of adding the junit output under the `if (defaultConfig.coverage) {` doesn't enable the junit reporting though. Therefore the pipeline won't be able to show the tests which have been executed in the `tests` tab view. I guess the coverage should be independent of the junit reporting. We only want to generate the junit reporting within the jenkins context.

So I'll rollback my last changes to be able to report to jenkins if no issues, besides, I need to clarify https://github.com/elastic/apm-agent-rum-js/pull/323#discussion_r303442179 as I want to confirm if the number of executed tests should be above 100 or not

Reply:
We have decided to keep the junit tests for tests running on Jenkins.